lib/bandit/clock.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Bandit.Clock do
+  @moduledoc false
+
+  use GenServer
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
```

Comment:
Naming this will preclude running multiple Bandit instances in the same system (which is a specific design goal of Bandit). I think the easiest way to solve this here is to host the Bandit.Clock process within the Bandit application (via `lib/bandit/application.ex` and an update to Bandit's mix.exs). It means that we'd lose the ability to specify date sending on a per-application basis, but I guess that's fine.